netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -324,12 +324,19 @@ private void cancelStream(ChannelHandlerContext ctx, CancelServerStreamCommand c
     encoder().writeRstStream(ctx, cmd.stream().id(), Http2Error.CANCEL.code(), promise);
   }
 
+  private static final ByteString FAST_CONTENT_TYPE = new ByteString(GrpcUtil.CONTENT_TYPE_GRPC,
+      US_ASCII);
+
   private void verifyContentType(int streamId, Http2Headers headers) throws Http2Exception {
     ByteString contentType = headers.get(CONTENT_TYPE_HEADER);
     if (contentType == null) {
       throw Http2Exception.streamError(streamId, Http2Error.REFUSED_STREAM,
           "Content-Type is missing from the request");
     }
+    // Avoid constructing a string and compare the bytes directly for fast path.
+    if (FAST_CONTENT_TYPE.equals(contentType)) {
```

Comment:
I wonder if we should find a way to make `GrpcUtil.isGrpcContentType` work properly for `AsciiString/ByteString` as well? Maybe change it to use `CharSequence`? Probably worth thinking about for a follow-up PR?